web/src/engine/main/osk/banner.ts

```diff
@@ -350,24 +482,79 @@ namespace com.keyman.osk {
       let rtl = activeKeyboard && activeKeyboard.isRTL;
       for (var i=0; i<SuggestionBanner.SUGGESTION_LIMIT; i++) {
         let indexToInsert = rtl ? SuggestionBanner.SUGGESTION_LIMIT - i -1 : i;
-        this.getDiv().appendChild(this.options[indexToInsert].div);
+        this.container.appendChild(this.options[indexToInsert].div);
 
-        if(i != SuggestionBanner.SUGGESTION_LIMIT) {
+        if(i != SuggestionBanner.SUGGESTION_LIMIT - 1) {
           // Adds a 'separator' div element for UI purposes.
           let separatorDiv = com.keyman.singleton.util._CreateElement('div');
           separatorDiv.className = 'kmw-banner-separator';
 
           let ds = separatorDiv.style;
-          ds.marginLeft = (SuggestionBanner.MARGIN / 2) + '%';
-          ds.marginRight = (SuggestionBanner.MARGIN / 2) + '%';
+          ds.marginLeft = `calc(${(SuggestionBanner.MARGIN / 2)}% - 0.5px)`;
+          ds.marginRight = `calc(${(SuggestionBanner.MARGIN / 2)}% - 0.5px)`;
 
-          this.getDiv().appendChild(separatorDiv);
+          this.container.appendChild(separatorDiv);
+          this.separators.push(separatorDiv);
         }
       }
+    }
 
-      this.manager = new SuggestionManager(this.getDiv(), this.options);
+    private updateOptions(suggestions: Suggestion[]) {
+      const fontStyle = getComputedStyle(this.options[0].div);
+      const emSizeStr = getComputedStyle(document.body).fontSize;
+      const emSize = getFontSizeStyle(emSizeStr).val;
 
-      this.setupInputHandling();
+      const textStyle = getComputedStyle(this.options[0].container.firstChild as HTMLSpanElement);
+
+      // TODO:  polish up; do a calculation that leaves perfect, clean edges when displaying exactly three options.
+      const targetWidth = this.width / 3; // Not fancy; it'll leave rough edges. But... it'll do for a demo.
+      const textLeftPad = new ParsedLengthStyle(textStyle.paddingLeft || '2px');   // computedStyle will fail if the element's not in the DOM yet.
+      const textRightPad = new ParsedLengthStyle(textStyle.paddingRight || '2px');
+
+      let optionFormat: OptionFormatSpec = {
+        paddingWidth: textLeftPad.val + textRightPad.val, // Assumes fixed px padding.
+        emSize: emSize,
+        styleForFont: fontStyle,
+        collapsedWidth: targetWidth,
+        minWidth: 0,
+      }
```

Comment:
Also tested with:

- `collapsedWidth: targetWidth, minWidth: targetWidth/2`
- `minWidth: targetWidth` (no collapsedWidth - so all suggestions got full length)
- `collapsedWidth: targetWidth, minWidth: targetWidth` (maintains the original banner's formatting)